### PR TITLE
Restructure et_relay subpackage

### PR DESCRIPTION
**Pull request by amaslenn**
## Summary
1. Make `et_replay` structure easier to refactor: added an entry point to export public symbols, used it in `train` code.
2. Enabled tests run as part of GHA CI.

## Test Plan
CI

## Additional Notes
—

### CODE REVIEW

**Comment by amaslenn** [comment]
cc @TaekyungHeo

File: et_replay/lib/__init__.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+from .execution_trace import ExecutionTrace
```
Comment (sanrise) [inline et_replay/lib/__init__.py:1]: +1 on exposing the ET class from the package __init__

File: et_replay/lib/comm/commsTraceParser.py [hunk 2]
```diff
@@ -2,14 +2,12 @@
...
 
-from et_replay.lib.comm import comms_utils
-from et_replay.lib.comm.comms_utils import commsArgs
-from et_replay.lib.comm.pytorch_backend_utils import supportedP2pOps
-
-from et_replay.lib.execution_trace import ExecutionTrace
+from .. import ExecutionTrace
+from . import comms_utils
+from .comms_utils import commsArgs
+from .pytorch_backend_utils import supportedP2pOps
```
Comment (sanrise) [inline et_replay/lib/comm/commsTraceParser.py:10]: can we keep the imports absolute? our absolute import strings are not really that long for us to justify relative imports. they also make code refactoring tricky. absolute imports are also more readable and pep8 friendly.

**Comment by sanrise** [comment]
The change is a good move to make et_replay more a package. But let's keep the imports absolute, they are not long enough yet for us to make them relative. Relative imports spoil readability, pep8 friendliness and refactoring capabilities. Within this single PR we are importing the `ExecutionTrace` class in the following manners:
`from .. import ExecutionTrace`, `from ..execution_trace import ExecutionTrace`, `from .execution_trace import ExecutionTrace`.

**Comment by amaslenn** [comment]
@sanrise thank you for the feedback! I mainly agree with you on relative imports topic, but there are a reason I made it like this:
1. Editors are going crazy: `et_replay` is a directory and at the same time it is a package. But the structures are different, `lib` doesn't exist in the package, but exist in file structure. So I need to either keep `lib` for both, or use relative imports.
2. While relative imports have bad reputation, they are all over python's code itself. And it kinda makes sense inside packages.

Anyway, I can use absolute imports and also keep `lib` and yet export some symbols from init. Seems fine to me. What do you think?

**Comment by sanrise** [comment]
> 1. Editors are going crazy: `et_replay` is a directory and at the same time it is a package. But the structures are different, `lib` doesn't exist in the package, but exists in the file structure. So I need to either keep `lib` for both, or use relative imports.

Sounds like presence of `lib` is vestigial at this point, and we can get rid of it. Without it, the content of `lib` moves up a level. This means 
* you don't need to patch the module path in your toml
* export what you want in your `__init__.py` to do `from et_replay import X, Y, Z`
* Your editor is happy.
* your absolute imports stick around.

Do you see a benefit of keeping it around just to have have it remapped in the toml? I don't, and I believe this will make everything more cleaner/easier imo.

**Comment by amaslenn** [comment]
> Do you see a benefit of keeping it around just to have have it remapped in the toml? I don't, and I believe this will make everything more cleaner/easier. Thoughts?

Generally I prefer having a top-level folder for a package or even have src-layout. It makes packaging more standard and manageable. But in our case IMO flat structure is fine too. I've updated my PR accordingly.

**Comment by sanrise** [comment]
> > Do you see a benefit of keeping it around just to have have it remapped in the toml? I don't, and I believe this will make everything more cleaner/easier. Thoughts?
> 
> Generally I prefer having a top-level folder for a package or even have src-layout. It makes packaging more standard and manageable. But in our case IMO flat structure is fine too. I've updated my PR accordingly.

thanks a lot! will import and run the linters / static checks shortly.

**Comment by TaekyungHeo** [comment]
@sanrise, please don't merge this PR until we inform you.

File: et_replay/utils.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
 import uuid
 from typing import Any, Dict
 
-from et_replay.lib.execution_trace import ExecutionTrace
+from et_replay.execution_trace import ExecutionTrace
```
Comment (sanrise) [inline et_replay/utils.py:9]: nit: for the record, this can also be `from et_replay import ExecutionTrace`. But this works too. (y)

Comment (amaslenn) [reply et_replay/utils.py]: Thank you! This makes sense, updated.

**Comment by TaekyungHeo** [comment]
@sanrise , you can continue code review at https://github.com/facebookresearch/param/pull/129

**Comment by briancoutinho** [comment]
@TaekyungHeo  after import we are seeing some Black formatting error. Did you guys run black formatter already?
If yes then we might need to disable the linter either outside or in meta, if you haven't i can share the format suggestions

**Comment by briancoutinho** [comment]
@TaekyungHeo also can you rebase to latest param main commit?

**Comment by amaslenn** [comment]
@sanrise shall we close this PR now?

**Comment by amaslenn** [comment]
Let's reopen if needed.

**Comment by TaekyungHeo** [comment]
@amaslenn, I believe we can close this PR as it has been merged by @sanrise. https://github.com/facebookresearch/param/pull/133

**Pull request closed without merging**